pyat/at/lattice/variable_elements.py

```diff
@@ -1,141 +1,259 @@
-            Seed(int): Seed of the random number generator for white
-                       noise excitation. Default datetime.now()
-            FuncA(list): User defined tbt kick list for PolynomA
-            FuncB(list): User defined tbt kick list for PolynomB
-            Periodic(bool): If True (default) the user defined kick is repeated
-            Ramps(list): Vector (t0, t1, t2, t3) in turn number to define the ramping 
-                         of the excitation
-
-              * ``t<t0``: excitation amplitude is zero
-              * ``t0<t<t1``: exciation amplitude is linearly ramped up
-              * ``t1<t<t2``: exciation amplitude is constant             
-              * ``t2<t<t3``: exciation amplitude is linearly ramped down
-              * ``t3<t``: exciation amplitude is zero           
+              * :py:attr:`.ACMode.ARBITRARY`: user defined turn-by-turn kick list
+        Keyword Arguments:
+            AmplitudeA(list,float): Amplitude of the excitation for PolynomA.
+                Default None
+            AmplitudeB(list,float): Amplitude of the excitation for PolynomB.
+                Default None
+            FrequencyA(float): Frequency of the PolynomA sine excitation. Unit Hz
+            FrequencyB(float): Frequency of the PolynomB sine excitation. Unit Hz
+            PhaseA(float): Phase of the sine excitation for PolynomA. Default 0 rad
+            PhaseB(float): Phase of the sine excitation for PolynomB. Default 0 rad
+            SinAabove(float): Limit the sin function to be above. Default -1.
+            SinBabove(float): Limit the sin function to be above. Default -1.
+            FuncA(list):   User defined tbt list for PolynomA
+            FuncB(list):   User defined tbt list for PolynomB
+            FuncAderiv1(list): tbt list for PolynomA 1st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncBderiv1(list): tbt list for PolynomB 1st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncAderiv2(list): tbt list for PolynomA 2st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncBderiv2(list): tbt list for PolynomB 2st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncAderiv3(list): tbt list for PolynomA 3st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncBderiv3(list): tbt list for PolynomB 3st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncAderiv4(list): tbt list for PolynomA 4st derivative wrt tau.
+                Default: zeros array of the custom function length.
+            FuncBderiv4(list): tbt list for PolynomB 4st derivative wrt tau.
+                Default: zeros array of the custom function length.
```

Comment:
The `list` type hint looks too restrictive: for me, it should rather be `Sequence[float]`,
with
```python
from collections.abc import Sequence
```